### ISSUE TITLE
Change how blog post "path" frontmatter works

### ISSUE THREAD

**Issue by benslv** [CyberSocYork/cybersoc.co.uk#50]
Currently, a blog post's frontmatter looks a little bit like this:

```md
---
path: "/blog/site-welcome"
date: "2020-04-16"
author: "Ben Silverman"
title: "Welcome to our new website!"
tags: ["post"]
---
```

with `path` telling Gatsby at what URL to put the blog post once it's been generated.

I don't really like how the `/path/` is included in there, because it means people could put posts at any URL they want on the site, rather than as a sub-page of the blog. I think it should be renamed to be the "slug" of the post (i.e. a unique, kebab-case string which identifies the post and will be used in the URL.

**TL;DR:** Gatsby should still place all blog posts at `/path/{slug}` but the `/path/` part should be moved to `gatsby-node.js` and handled at build time.

**Comment by Sciencentistguy** [comment]
I don't completely love the way the tags are shown. It's a bit in-your-face.

Could we have something like how gmail does it? Smaller, and possible next to or just under the title?

![image](https://user-images.githubusercontent.com/4983935/140657520-0388dc93-c031-4d0c-a9c5-47fc3745ede0.png)

Also the "view all tags" page needs a little bit more styling IMO.
